Services.Test/Storage/StorageRecordsTest.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Concurrency;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.DataStructures;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage.DocumentDb;
+using Moq;
+using Xunit;
+
+namespace Services.Test.Storage
+{
+    public class StorageRecordsTest
```

Comment:
I think we need more tests to cover failures

Reply:
Added:

- ItLogsErrorWhenTryingToDeleteARecordThatDoesNotExist
- ItThrowsConflictingResourceExceptionWhenCreatingAResourceWithAnIdThatIsInUse
- ItLogsAnErrorWhenADbOperationFails